cunumeric/array.py

```diff
@@ -2994,6 +2994,31 @@ def reshape(self, shape, order="C") -> ndarray:
         --------
         Multiple GPUs, Multiple CPUs
         """
+
+        computed_shape = (shape,) if isinstance(shape, int) else shape
+
+        if sum(extent == -1 for extent in computed_shape) > 1:
```

Comment:
NumPy will accept any negative number as an "unknown" dimension, so I would change this to
```
extent < 0 for extent in computed_shape
```